test/test_namedtensor.py

```diff
@@ -272,6 +272,21 @@ def test_stride(self):
         with self.assertRaisesRegex(RuntimeError, 'Name \'N\' not found in '):
             torch.empty(2, 3, 4).stride('N')
 
+    def test_transpose_variants(self):
+        t = torch.randn(2, 3, 5, 7, names=('N', 'C', 'H', 'W'))
+        self.assertEqual(t.transpose('N', 'C').names, ['C', 'N', 'H', 'W'])
+        self.assertEqual(t.transpose(1, 3).names, ['N', 'W', 'H', 'C'])
+
+        t = torch.randn(2, 3, names=('N', 'C'))
+        self.assertEqual(t.t().names, ['C', 'N'])
+
+    @unittest.expectedFailure
```

Comment:
what's expected to fail?

Reply:
In-place transpose ops. I'll make the test name clearer.

Reply:
fail with an error or the assertions should fail? both fail? I believe a failed assertion raises an exception, so if you're expecting the first assertion to fail I'd think the second one wouldn't get run.

If it's the assertion, I'd think you could assert not equal?

Reply:
Eh, It might be better to just delete this test. Maintaining it is more trouble than it's worth, and it's meant to be a TODO for me.

Reply:
what is it that doesn't work or fails though? maybe an assertRaises would make sense?

Reply:
I'm fine leaving the test out if it doesn't make sense, I just want to understand what is/isn't supported. wasn't clear to me from the (now removed) test

Reply:
What isn't supported are t_ and transpose_. So the right way to write the tests would have been to have 2 expected failures, one for each of them.

I don't want to do an assertRaises because that gives the impression that it is intentional that both of them are not supported. It isn't intentional, I just haven't had the time to implement them correctly.

Since the pull request title only talks about t() and transpose(), and not about the in-place variants, the current tests should be sufficient.

Reply:
ah OK.

It's fine to leave the tests out, but I think it is sort of conventional in our codebase to add assertions of failure for features we intend to add later. It sort of points us to and reminds us of where the tests go when we add the functionality later.

Reply:
>  I think it is sort of conventional in our codebase to add assertions of failure for features we intend to add later.

Sure. However, I'd like to come up with a good plan for doing assertions like that before committing to it. Having all the unimplemented ops spread over multiple tests is not the best, so it would be nice to gather them all in one table or something.